app/src/main/java/org/mozilla/focus/exceptions/GeckoExceptionsMigrator.kt

```diff
@@ -0,0 +1,71 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.exceptions
+
+import android.content.Context
+import android.content.SharedPreferences
+import mozilla.components.support.base.log.logger.Logger
+import org.mozilla.geckoview.GeckoRuntime
+import org.mozilla.geckoview.GeckoSession.PermissionDelegate.ContentPermission.VALUE_ALLOW
+import org.mozilla.geckoview.GeckoSession.PermissionDelegate.PERMISSION_TRACKING
+
+/**
+ * Migrate preference based Focus tracking protection exceptions to gecko based exceptions.
+ */
+class GeckoExceptionsMigrator(
+    private val runtime: GeckoRuntime
+) {
+    private val logger = Logger("GeckoExceptionsMigrator")
+
+    fun start(context: Context) {
+        if (!isOver(context)) {
+            logger.info("Starting tracking protection exceptions migration")
+            val exceptions = fetchExceptions(context)
+            exceptions.forEach {
+                migrate(it)
+            }
+            logger.info("Finished tracking protection exceptions migration")
+            preferences(context).edit().remove(KEY_DOMAINS).apply()
+        }
+    }
+
+
+    /**
+     * Fetch the previously added/saved custom domains from preferences.
+     *
+     * @param context the application context
+     * @return list of custom domains
+     */
+    private fun fetchExceptions(context: Context): List<String> {
+        return (preferences(context)
+                .getString(KEY_DOMAINS, "") ?: "")
+                .split(SEPARATOR)
+                .filter { it.isNotEmpty() }
+
+    }
+
+    private fun isOver(context: Context) = preferences(context).contains(PREFERENCE_NAME)
+
+    private fun preferences(context: Context): SharedPreferences =
+            context.getSharedPreferences(PREFERENCE_NAME, Context.MODE_PRIVATE)
+
+    private fun migrate(uri: String) {
+        val privateMode = true
```

Comment:
nit: I can we can just put `true` into the function call below and do not need the `privateMode` reference?

Reply:
Yeah, 🙃  thanks.

Reply:
Yes, I just wanted to be explicit about what it's this parameter about, as this is java api we can't use named parameters,